src/Symfony/Contracts/HttpClient/HttpClientInterface.php

```diff
@@ -48,11 +48,9 @@ interface HttpClientInterface
         'buffer' => true,       // bool|resource|\Closure - whether the content of the response should be buffered or not,
                                 //   or a stream resource where the response body should be written,
                                 //   or a closure telling if/where the response should be buffered based on its headers
-        'on_progress' => null,  // callable(int $dlNow, int $dlSize, array $info, ?Closure $resolve = null) - throwing any
-                                //   exceptions MUST abort the request; it MUST be called on connection, on headers and on
-                                //   completion; it SHOULD be called on upload/download of data and at least 1/s;
-                                //   if passed, $resolve($host) / $resolve($host, $ip) can be called to read / populate
-                                //   the DNS cache respectively
+        'on_progress' => null,  // callable(int $dlNow, int $dlSize, array $info) - throwing any exceptions MUST abort the
+                                //   request; it MUST be called on connection, on headers and on completion; it SHOULD be
+                                //   called on upload/download of data and at least 1/s
```

Comment:
Reverting to the previous (before the latest CVE) signature for the progress function: it's way more seamless to expose the resolve function as an implementation detail using the info array.